src/plugins/chart_expressions/expression_partition_vis/public/utils/layers/sort_predicate.ts

```diff
@@ -34,8 +34,9 @@ const sortPredicateSaveSourceOrder: SortPredicatePureFn =
   () =>
   ([, node1], [, node2]) => {
     const [index1] = node1.inputIndex ?? [];
+    // Sorting for first group (we should respect the order, which is why we return 0)
     if (index1 !== undefined) {
-      return index1;
+      return 0;
```

Comment:
As discussed offline, this shouldn't assume the array is in the right order already but instead should compare the indices of node1 and node2 (same for the other sorting predicate)